File: py/max_value/solution_test.py
import unittest
from solution import Solution

class TestMaxValue(unittest.TestCase):
    def setUp(self):
        self.solution = Solution()

    def test_case_one(self):
        nums = [4, 7, 2, 8, 10, 9]
        expected = 10
        result = self.solution.max_value(nums)
        self.assertEqual(result, expected)

    def test_case_two(self):
        nums = [10, 5, 40, 40.3]
        expected = 40.3
        result = self.solution.max_value(nums)
        self.assertEqual(result, expected)

    def test_case_three(self):
        nums = [-5, -2, -1, -11]
        expected = -1
        result = self.solution.max_value(nums)
        self.assertEqual(result, expected)

    def test_case_four(self):
        nums = [42]
        expected = 42
        result = self.solution.max_value(nums)
        self.assertEqual(result, expected)

    def test_case_five(self):
        nums = [1000, 8]
        expected = 1000
        result = self.solution.max_value(nums)
        self.assertEqual(result, expected)

    def test_case_six(self):
        nums = [1000, 8, 9000]
        expected = 9000
        result = self.solution.max_value(nums)
        self.assertEqual(result, expected)

    def test_case_seven(self):
        nums = [2, 5, 1, 1, 4]
        expected = 5
        result = self.solution.max_value(nums)
        self.assertEqual(result, expected)

File: java/maxValue/src/test/java/dev/jgn/soution/SolutionTest.java
package dev.jgn.soution;

import dev.jgn.solution.Solution;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class SolutionTest {

    @Test
    void testCaseOne() {
        var solution = new Solution();
        var nums = new int[]{4, 7, 2, 8, 10, 9};
        var result = solution.maxValue(nums);
        var expected = 10;
        assertEquals(expected, result);
    }

    @Test
    void testCaseTwo() {
        var solution = new Solution();
        var nums = new int[]{10, 5, 40, 42};
        var result = solution.maxValue(nums);
        var expected = 42;
        assertEquals(expected, result);
    }

    @Test
    void testCaseThree() {
        var solution = new Solution();
        var nums = new int[]{-5, -2, -1, -11};
        var result = solution.maxValue(nums);
        var expected = -1;
        assertEquals(expected, result);
    }

    @Test
    void testCaseFour() {
        var solution = new Solution();
        var nums = new int[]{42};
        var result = solution.maxValue(nums);
        var expected = 42;
        assertEquals(expected, result);
    }

    @Test
    void testCaseFive() {
        var solution = new Solution();
        var nums = new int[]{1000, 8};
        var result = solution.maxValue(nums);
        var expected = 1000;
        assertEquals(expected, result);
    }

    @Test
    void testCaseSix() {
        var solution = new Solution();
        var nums = new int[]{1000, 8, 9000};
        var result = solution.maxValue(nums);
        var expected = 9000;
        assertEquals(expected, result);
    }

    @Test
    void testCaseSeven() {
        var solution = new Solution();
        var nums = new int[]{2, 5, 1, 1, 4};
        var result = solution.maxValue(nums);
        var expected = 5;
        assertEquals(expected, result);
    }

}
